TrackingTools/GsfTracking/src/TsosGaussianStateConversions.cc

```diff
@@ -40,12 +40,15 @@ namespace GaussianStateConversions {
     std::vector<TrajectoryStateOnSurface> components;
     components.reserve(singleStates.size());
     for (auto const& ic : singleStates) {
-      components.emplace_back((*ic).weight(),
-                              LocalTrajectoryParameters((*ic).mean(), pzSign, charged),
-                              LocalTrajectoryError((*ic).covariance()),
-                              surface,
-                              field,
-                              side);
+      //require states to be positive-definite
+      if (double det = 0; (*ic).covariance().Det2(det) && det > 0) {
```

Comment:
one could ask why using this simplified criterion here, and another one in https://github.com/cms-sw/cmssw/blob/83e46c02e21a9d83b997f1c19a93b35b8695e689/TrackingTools/GsfTracking/src/GsfMultiStateUpdator.cc#L34-L39

Reply:
yes, as we were discussing also in the [github-issue](https://github.com/cms-sw/cmssw/issues/39570), indeed the checks for positive definiteness are not harmonised across different modules. The check in [GsfMultiStateUpdator.cc](https://github.com/cms-sw/cmssw/blob/83e46c02e21a9d83b997f1c19a93b35b8695e689/TrackingTools/GsfTracking/src/GsfMultiStateUpdator.cc#L34-L39) module became more and more strict over time to address various issues and crashes. 

In this PR, I've tried to address only one crash in the event, that was posted in the T0 cmsTalk. But in general, I do see the room for improvements in GSF code.

Reply:
I think given the urgency, let's adjust the rest at a later time. I've made a GitHub issue https://github.com/cms-sw/cmssw/issues/39933